packages/dashboard/components/consent/pages/index.tsx

```
@@ -165,186 +216,76 @@ export default function IndexView() {
     };
   }, [sessionSigs]);
 
-  // Unified loading state
-  if (authLoading || accountsLoading || sessionLoading) {
-    return (
-      <CenteredContainer>
+  // ------ RENDER CONTENT ------
+  
+  const renderContent = () => {
+    // Handle loading states first
+    if (authLoading || accountsLoading || sessionLoading || isProcessing) {
+      return (
         <Loading 
           copy={loadingMessage} 
           isTransitioning={isTransitioning}
         />
-      </CenteredContainer>
-    );
-  }
-
-  // Authenticated states
-  if (userPKP && sessionSigs) {
-    // Save the PKP info in localStorage for SessionValidator to use
-    try {
-      updateAuthInfo({
-        agentPKP,
-        userPKP
-      });
-    } catch (error) {
-      console.error('Error saving PKP info to localStorage:', error);
+      );
     }
-
-    return (
-      <CenteredContainer>
+    
+    // If we have existing auth info, show the option to use it
+    if (showExistingAccount) {
+      return (
+        <ExistingAccountView
+          authInfo={authInfo}
+          handleUseExistingAccount={handleUseExistingAccount}
+          handleSignOut={handleSignOut}
+        />
+      );
+    }
+  
+    // If authenticated with a new PKP and session sigs
+    if (userPKP && sessionSigs) {
+      // Save the PKP info in localStorage for SessionValidator to use
+      try {
+        updateAuthInfo({
+          agentPKP,
+          userPKP
+        });
+      } catch (error) {
+        console.error('Error saving PKP info to localStorage:', error);
```

Is it _really_ not a critical error if calling `updateAuthInfo()` throws a hard runtime error? This is swallowing the failure entirely, but it feels like if this fails, their authInfo is in a totally inconsistent state and they may need to re-log in, right?

---

Changed to have `updateAuthInfo` include an error which will be returned, currently shown in a popup (for now):
```
catch (error) {
        console.error('Error saving PKP info to localStorage:', error);
        showError(error as Error, 'Authentication Error');
        return (
          <LoginMethods
            authWithEthWallet={authWithEthWallet}
            authWithWebAuthn={authWithWebAuthn}
            authWithStytch={authWithStytch}
            registerWithWebAuthn={handleRegisterWithWebAuthn}
          />
        );
      }
```

---

I guess we can also remove the `lit-auth-info` if this fails?